### PR TITLE
VIH-9672 Upgrade to Angular 15

### PR DESCRIPTION
**Pull request by oliver-scott**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/VIH-9672

### Change description ###
- Upgrade Angular and associated packages to v15
- Convert tslint to eslint (tslint now deprecated)
- Fixes for lint errors not previously showing in tslint
- Renamed tsconfig.base.json to tsconfig.json and removed tsconfig.json. Required for installing eslint
- Update sonar cloud coverage exclusion paths

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: AdminWebsite/AdminWebsite/AdminWebsite.csproj [hunk 1]
```diff
@@ -93,13 +90,11 @@
 	<Target Name="PublishRunWebpack" AfterTargets="ComputeFilesToPublish" Condition=" '$(PublishRunWebpack)' == 'true' ">
 		<!-- As part of publishing, ensure the JS resources are freshly built in production mode -->
 		<Exec WorkingDirectory="$(SpaRoot)" Command="npm install" />
-		<Exec WorkingDirectory="$(SpaRoot)" Command="npm run build -- --prod" />
-		<Exec WorkingDirectory="$(SpaRoot)" Command="npm run build:ssr -- --prod" Condition=" '$(BuildServerSideRenderer)' == 'true' " />
+		<Exec WorkingDirectory="$(SpaRoot)" Command="npm run build-prod" />
```
Comment (oliver-scott) [inline AdminWebsite/AdminWebsite/AdminWebsite.csproj:93]: note: the npm run build:ssr command does not work in angular 15. Checked and could not see any references to BuildServerSideRenderer, so have removed it

File: AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/edit-work-hours/vho-work-hours-non-availability-table/vho-work-hours-non-availability-table.component.ts [hunk 2]
```diff
@@ -91,12 +91,6 @@ export class VhoWorkHoursNonAvailabilityTableComponent implements OnInit, CanDea
     @Output() cancelSaveNonWorkHours: EventEmitter<void> = new EventEmitter();
     showSaveConfirmation = false;
 
-    checkType(myArray: any[], type: any): boolean {
```
Comment (oliver-scott) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/edit-work-hours/vho-work-hours-non-availability-table/vho-work-hours-non-availability-table.component.ts:94]: note: unused function, removed

File: AdminWebsite/AdminWebsite/ClientApp/src/app/booking/hearing-schedule/hearing-schedule.component.html [hunk 3]
```diff
@@ -172,7 +172,7 @@ <h2 class="govuk-heading-m">Time and location</h2>
               id="hearingStartTimeMinute"
               name="hearingStartTimeMinute"
               type="number"
-              min="1"
+              min="0"
```
Comment (oliver-scott) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/booking/hearing-schedule/hearing-schedule.component.html:175]: note: had to change this as when to set 1, you can't book hearings on the hour in angular 15 (15:00, 16:00 etc)

**Review by will-craig — APPROVED** [review]

**Comment by will-craig** [comment]
Looks good to me, it's such a large change wouldn't blame you for wanting a second pair of eyes on it aswell though,

**Review by marcogagliardi — APPROVED** [review]
👌🏽

**Pull request merged**